tests/integration/security/pass_through_filter_chain_test.go

```diff
@@ -166,13 +238,18 @@ func TestPassThroughFilterChain(t *testing.T) {
 							if len(responses) < 1 {
 								return fmt.Errorf("received no responses from request to %s", host)
 							}
-							if response.StatusCodeOK != responses[0].Code {
+							if tc.schema == protocol.HTTP && response.StatusCodeOK != responses[0].Code {
 								return fmt.Errorf("want status %s but got %s", response.StatusCodeOK, responses[0].Code)
 							}
-						} else if err == nil || !strings.Contains(err.Error(), "EOF") {
-							// The request should always be rejected in TCP level because we currently do not support
-							// HTTP filters in pass through filter chains.
-							return fmt.Errorf("want error EOF but got: %v", err)
+						} else {
+							// Check HTTP forbidden response
```

Comment:
Is this part for testing a connection rejected in TCP layer? If so, why do we need to check the `responses`?

Reply:
It also checks HTTP rejection. This branch is for `want: false`